include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -389,35 +389,31 @@ struct __gen_set_mask
 
         std::size_t __nb = __set_b.size();
 
-        const auto& __val_a_proj = std::invoke(__proj1, __set_a[__id]);
```

Comment:
We are fixing a case with a proxy, right?

Instead of calling `invoke` with arguments multiple times, we can use `auto&&`. If its a proxy value, then `__set_a[__id]` returns a value and `__val_a_proj` will be a copy. Otherwise it will be a reference (in the most cases), hence no copy-constructor is needed.

Reply:
This proposal has been implemented in the PR https://github.com/uxlfoundation/oneDPL/pull/2460